examples/blackjack.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+This is a blackjack game made with pyCardDeck, meant as an example rather than
+as a complete game.
+"""
+
+import sys
+import pyCardDeck
+from typing import List
+from pyCardDeck.cards import PokerCard
+
+# Copy-pasted from examples/poker.py
+class Player:
+
+    def __init__(self, name: str):
+        self.hand = []
+        self.name = name
+
+    def __str__(self):
+        return self.name
+
+class BlackjackGame:
+
+    def __init__(self, players: List[Player]):
+        self.deck = pyCardDeck.Deck(
+            cards=generate_deck(),
+            name='Poker deck',
+            reshuffle=False)
+        self.players = players
+        self.scores = {}
+        print("Created a game with {} players.".format(len(self.players)))
+
+    def blackjack(self):
+        """
+        The main blackjack game sequence.
+
+        Each player takes an entire turn before moving on.
+
+        If each player gets a turn and no one has won, the player or players
+        with the highest score below 21 are declared the winner.
+        """
+        print("Setting up...")
+        print("Shuffling...")
+        self.deck.shuffle()
+        print("All shuffled!")
+        print("Dealing...")
+        self.deal()
+        print("\nLet's play!")
+        for player in self.players:
+            print("{}'s turn...".format(player.name))
+            self.play(player)
+        else:
+            print("That's the last turn. Determining the winner...")
+            self.find_winner()
+
+    def play(self, player):
+        """
+        An individual player's turn.
+
+        If the player's cards are an ace and a ten or court card,
+        the player has a blackjack and wins.
+
+        If a player's cards total more than 21, the player loses.
+
+        Otherwise, it takes the sum of their cards and determines whether
+        to hit or stand based on their current score.
+        """
+        while True:
+            points = sum_hand(player.hand)
+            if points < 17:
+                print("   Hit.")
+                self.hit(player)
+            elif points == 21:
+                print("   {} wins!".format(player.name))
+                sys.exit(0) # End if someone wins
+            elif points > 21:
+                print("   Bust!")
+                break
+            else:  # Stand if between 17 and 20 (inclusive)
+                print("   Standing at {} points.".format(str(points)))
+                self.scores[player.name] = points
+                break
+
+    def find_winner(self):
+        """
+        Finds the highest score, then finds which player(s) have that score,
+        and reports them as the winner.
+        """
+        winners = []
+        try:
+            win_score = max(self.scores.values())
+            for key in self.scores.keys():
```

Comment:
The players' score are kept in a dictionary of `{player_name: score}` pairs. The purpose of using `self.scores.keys()` was so that when the correct score was found, its key could be added to the winners list. 

I see that accessing a score with a key is less efficient than referencing the score directly, but I don't know of another way to access the key via the score, barring something like predicate calculus (times like this make me wish I were better at Prolog...)